sklearn/base.py

```diff
@@ -370,18 +370,16 @@ def _check_n_features(self, X, reset):
 
         if reset:
             self.n_features_in_ = n_features
-        else:
-            if not hasattr(self, 'n_features_in_'):
-                raise RuntimeError(
-                    "The reset parameter is False but there is no "
-                    "n_features_in_ attribute. Is this estimator fitted?"
-                )
-            if n_features != self.n_features_in_:
-                raise ValueError(
-                    'X has {} features, but {} is expecting {} features '
-                    'as input.'.format(n_features, self.__class__.__name__,
-                                       self.n_features_in_)
-                )
+            return
+
+        fitted_n_features_in = getattr(self, 'n_features_in_', None)
+        if fitted_n_features_in is None:
+            return
```

Comment:
I think that `hasattr` would be cleaner, especially since `self.n_features_in_` is used later and `fitted_n_features_in` isn't